src/core/reactor.cc

```diff
@@ -2965,6 +2965,35 @@ void reactor::add_urgent_task(task* t) noexcept {
     }
 }
 
+void reactor::run_in_background(future<> f) {
+    if (f.available()) {
+        if (f.failed()) {
+            std::rethrow_exception(f.get_exception());
+        }
+        return;
+    }
```

Comment:
It throws if gate is closed. I'm not sure we want to disrupt the application in this case since by not waiting on the background fiber the caller indicates they don't care if it succeeds or not.
Otherwise, they must wait on it.